bundle/jsky.app.ot/src/main/scala/jsky/app/ot/ags/BagsManager.scala

```diff
@@ -129,37 +151,44 @@ final class BagsManager(executor: ScheduledThreadPoolExecutor) {
                       // was running, so discard the result.
                       if (!state.keys(key)) {
                         LOG.info(s"$bagsIdMsg successful. Results=${optExtract(selOpt) ? "Yes" | "No"}.")
-                        if (ObservationStatus.computeFor(obs) != ObservationStatus.OBSERVED) {
-                          Swing.onEDT {
-                            obs.getProgram.removeCompositeChangeListener(CompositePropertyChangeListener)
-                            obs.getProgram.removeStructureChangeListener(StructurePropertyChangeListener)
-                            worker(TpeContext(obs), selOpt)
-                            obs.getProgram.addStructureChangeListener(StructurePropertyChangeListener)
-                            obs.getProgram.addCompositeChangeListener(CompositePropertyChangeListener)
-                          }
+                        Swing.onEDT {
+                          obs.getProgram.removeCompositeChangeListener(CompositePropertyChangeListener)
+                          obs.getProgram.removeStructureChangeListener(StructurePropertyChangeListener)
+                          worker(TpeContext(obs), selOpt)
+                          obs.getProgram.addStructureChangeListener(StructurePropertyChangeListener)
+                          obs.getProgram.addCompositeChangeListener(CompositePropertyChangeListener)
                         }
+                        bagsClearStatus()
                       }
 
+                    // NOTE: We sleep for a brief duration BEFORE enqueue to allow the error message to be readable.
                     // We don't want to print the stack trace if the host is simply unreachable.
                     // This is reported only as a GenericError in a CatalogException, unfortunately.
```

Comment:
Probably too late to improve this, but would you benefit by the catalog producing a more specific error if the host is unreachable?